timon/state.py

```diff
@@ -295,38 +295,38 @@ def has_state_changed(self, probe, status, flap_detection=True,
             previous_rslt_to_compare = pst[-flap_cnt]
             return previous_rslt_to_compare[1] != status
 
-    async def update_probe_state(
-            self, probe,
-            status, msg=None, t=None, save=False):
+    async def update_probe_state(self, probe, status, msg, t):
         """ updates a probe state
-            :param save: will also be saved to disk for potential
-                    web status / notifiers / etc.
-
-            :returns True if state changed
+            called from `probe_done` (which is probs' `done_cb`)
         """
-        t = t or time.time()
         probe_name = probe.name
-        if save:
-            raise NotImplementedError("save option is still not implemented")
+
         has_state_changed_wo_flap_detection = self.has_state_changed(
             probe=probe, status=status, flap_detection=False)
 
         prb_states = self.state['probe_state']
         if probe_name not in prb_states:
             prb_states[probe_name] = []
         pst = prb_states[probe_name]
-        prev_status = pst[-1][1] if pst else "UNKNOWN"
         pst.append((t, status, msg))
+        # note to self: slice assignment _through a reference!_ ie even
+        # if `pst` isn't used here afterward, it does mutate the underlying
+        # list which is stored in `self.state['probe_state'][probe_name]`!
```

Comment:
this note can be useful.

I don't understand however the.

if `pst` isn't used here afterwards.



A potential suggestion for the comment:

assigning to a slice does not create a new list instance it modifies the existing one and all references to the previous list object are preserved.